kernel/serial_port/src/lib.rs

```diff
@@ -66,6 +66,9 @@ static COM2_SERIAL_PORT: Once<Arc<MutexIrqSafe<SerialPort>>> = Once::new();
 static COM3_SERIAL_PORT: Once<Arc<MutexIrqSafe<SerialPort>>> = Once::new();
 static COM4_SERIAL_PORT: Once<Arc<MutexIrqSafe<SerialPort>>> = Once::new();
 
+pub enum SerialPortError {
+    AlreadyExists,
+}
```

Comment:
this error/enum variant name is misleading. It sounds like it'd be used to indicate that the serial port already exists, but instead it is actually used to indicate that a serial port already had a data sender. It should be named accordingly.